contracts/near/eth-client/src/tests.rs

```diff
@@ -574,6 +574,89 @@ fn bsc_validate_headers() {
     assert!(contract.headers.len() == 10);
 }
 
+#[test]
+#[cfg_attr(not(feature = "bsc"), ignore)]
+fn bsc_validate_headers_genesis() {
+    testing_env!(get_context(vec![], false));
+    let (blocks, hashes) = get_blocks(&BSC_WEB3RS, 0, 10);
+    let chain_id = 97;
+    
+    let mut contract = EthClient::init(
+        true,
+        String::from("bsc"),
+        0,
+        vec![],
+        blocks[0].clone(),
+        210,
+        210,
+        210,
+        None,
+        chain_id,
+    );
+
+    for block in blocks.into_iter().skip(1) {
+        contract.add_block_header(block, vec![]);
+    }
+    assert!(hashes[0] == contract.epoch_header);
+    assert!(contract.headers.len() == 10);
+}
```

Comment:
This is not relevant anymore after #679, this should be either negative or removed

Reply:
Fixed in e6e88c